### ISSUE TITLE
Possibility to enter the battery pack number to keep track of user batteries performance

### ISSUE THREAD

**Issue by bsongis-frsky** [FrSkyRC/ETHOS-Feedback-Community#607]
Petite suggestion d’amélioration au niveau des radios en ce qui concerne le vol en électrique. Il serait bien de pouvoir saisir avant chaque vol un numéro d’accu qui serait enregistré dans les fichiers log. Ceci permettrait de vérifier les performances des accus (enregistrement par module FLVS) dans le temps. Actuellement il est impossible sauf à se rappeler de l’ordre d’utilisation des accus d’avoir cette info.

A bientôt et bel été.

Régis

**Comment by bsongis-frsky** [comment]
Waiting for ideas to implement this correctly

**Comment by loetefix** [comment]
Hi,
This is a realy nice idea.
We need the possibbility to create a lipo database with 

LipoNumber  - Date of first use  -  Cells  -  Capacity(mah)   -  C-Rate  - Manufacturer    -   Name  - total used capacity(autoCalc by ethos)

And a widget where you can choose the Lipo you`re using.

**Comment by bsongis-frsky** [comment]
Yes this is the original idea from Regis, I am sure someone will design some nice UI @ceeb182 ;)

**Comment by jreis1** [comment]
I think this is an interesting request that will allow Ethos/frsky devices to stand out from the competitors. 

For systems like RB30/40, I think this would be easy to do as they collect plenty information (voltage, current, maybe internal resistance?) from the power source that may allow somehow to characterize them. 

But keeping in mind that most people just uses a standard RX, and only Vbat is provided, doing this via telemetry info would be difficult. 

I agree with @loetefix and RFID would be the easiest way to keep track off all the batteries (with the aid of a database), with reduced cost (RFID tags are pretty cheap and super lightweight). 

However, this would require some hardware changes in the Tx (to incorporate a RFID reader).  To facilitate this and avoid hardware changes, some external RFID reader could be created.

This can be done by adding RFID reading features to new receivers, by creating a RFID receiver that will connect to any existing receiver via smartport and transmit data via telemetry, or create a standalone RFID reader that will transfer data to ethos via Bluetooth.

**Comment by loetefix** [comment]
my idea was a rfid reader for attach to the smartport ;) That seems to be preaty easy.

**Comment by jreis1** [comment]
> my idea was a rfid reader for attach to the smartport ;) That seems to be preaty easy.

Agreed! Care must be taken when selecting the RFID reader as there are a few operating in the 900MHz band. Mostly though are in 13.56 MHz.

**Comment by loetefix** [comment]
dont know about interference with 900mhz. but it should be no problem to activate the system only short after connecting the battery.

**Comment by Hobby4life** [comment]
This alone would be nice for persistant consumption use of lipo's.

**Comment by andika207** [comment]
> A bientôt et bel été.

@bsongis-frsky  bonnes vacances! vous méritez un repos pour recharger les piles.

**Comment by Ceeb182** [comment]
I'm having a little trouble with all the projects on the go at the moment.
For the management of batteries, we must distinguish several types of equipment and therefore several types of use:
- Scenario 1 : use of the analog input AIN2 with voltage divider to get the global voltage of the battery
- Scenario 2 : use of a FLVSS which allows to obtain the voltage of each cell + global voltage
- Scenario 3 : use of a FLVSS + current sensor (FAS or RB) allowing in addition to obtain the real consumption  

It seems essential to me, in addition to the ID of the battery used, to link it to a battery base which will be in the form of a Log file dedicated to the batteries (thus nothing to store in RAM other than the ID of the battery).
This special battery Log file must allow two things :
- identify the battery and its characteristics
- to make the log of each alarm reached (example: Voltage too low, DeltaV too high, Voltage full charge reached, Capacity consumed) with reference to the model log if it exists.  

On the developer side, the advantage of having a dedicated log allows to easily retrieve a parameter already used on another model without excessive use of RAM (example: for the battery #A retrieve the capacity consumed on model 1 + capacity consumed for model 2)
On the user's side, it will be much easier to retrieve the data concerning a particular battery: the main part is in the Log file dedicated to the batteries, the details are in the respective Logs of the models used.

Beyond battery management, we can imagine (much later...) a similar system for other "external resources" such as heat engines, turbojet engines, complex drive trains, etc... in short any resource that needs to be monitored to analyze its wear or to trigger preventive maintenance. Even if this remains a concept for the moment, it makes me say that it will probably be necessary to have a new "Third-party resource" menu to manage what needs to be monitored and the different alarm thresholds. It is also with the "Third-party resource" menu that we will be able to visualize and analyse the data in detail for the monitored resource. Note that the default monitoring rules will be proposed by a wizard (in the same way as the wizard for creating a new model), and can be modified manually (in the same way as the mixers of a model).

For the graphical aspect in the widgets, I imagine a visual that includes several types of display in the same widget: value and/or thermometer-type alarm and/or graph and/or message and/or bar graph. The "Third-party resource" widget does not display any details, it is there to remind the main elements and main alarms. The Widget is also a shortcut to select the battery (or resource) being monitored without going through a more detailed menu like "Third-party resource".

For the physical management aspect, the coding of the ID of an external resource must be simple and understandable. A simple indelible pen should be enough to reference its resource. We can imagine the ID of Bertrand SONGIS's battery in the following way: BESO-B001 (First name + Last name + B for "Batt" + order number), which allows to differentiate it from Celestin EBRARD: CEEB-B001.

This general vision should allow the developer to imagine a "third-party resource" management engine ready to manage different resources in the long term. 
Of course, it is necessary to **start simple** by managing for the moment only the batteries on some established criteria with a rudimentary graphic interface.

To be continued in mid-August...

**Comment by loetefix** [comment]
keep alive

**Comment by andika207** [comment]
ce sujet commence par @bsongis-frsky et maintenant le bot de github le ferme puisque il n' a plus aucun interêt ???
c'est vraiment bizarre de vouloir un bot dans ce repo pour se débarasser du travail au lieu de laisser le sujet ouvert pour plus tard.
